src/main/java/jenkins/branch/BranchEventCause.java

```diff
@@ -49,10 +49,13 @@
 
     private final String description;
 
+    private final SCMEvent event;
+
     BranchEventCause(@NonNull SCMEvent<?> event, @CheckForNull String description) {
         timestamp = event.getTimestamp();
         origin = event.getOrigin();
         this.description = StringUtils.isBlank(description) ? event.description() : description;
+        this.event = event;
```

Comment:
Since the event has to be `@CheckForNull`...
```suggestion
        this.event = event instanceof Serializable ? event : null;
```